pkgs/development/tools/loganalyzer/default.nix

```diff
@@ -0,0 +1,62 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, qtbase
+, qtsvg
+, wrapQtAppsHook
+}:
+
+let
+  pname = "loganalyzer";
+  appname = "LogAnalyzer";
+  version = "23.5.1";
+in
+stdenv.mkDerivation {
+  inherit pname appname version;
+
+  src = fetchFromGitHub {
+    owner = "pbek";
+    repo = "loganalyzer";
+    rev = "v${version}";
+    fetchSubmodules = true;
```

Comment:
Can we use the qmarkdowntextedit from nixpkgs instead?

Reply:
(Hah, I didn't even know someone packaged it. 😁)
Hm, that would be unreliable, because I only test with the version I pin the sub-module to. There could be breaking API changes.

Reply:
There hardly is any difference, qmarkdowntextedit is very small.
Before I have created and used a source file with all the code in it, but @natsukium was against it, so we settled at using the submodule (which is fine with me). The submodule code is part of the code. There are no other external dependencies (other than Qt).
Ripping that out means the qmake project file needs patching...

Reply:
QOwnNotes for example has plenty of submodules. All of them are part of the source archive that is used in the [default.nix](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/office/qownnotes/default.nix)...

Reply:
> fetching submodules makes that src less reproducible over time.

so I can understand it better... is that really the case? all git submodules are each pinned to a commit hash of the included repo...

Reply:
> so I can understand it better... is that really the case? all git submodules are each pinned to a commit hash of the included repo...

Some metadata sometimes already changed over the past as it uses git to download the files rather than downloading the archive from github and then unzipping that. Generally it is fine and not as bad as leaveDotGit but it should be avoided if easily possible.

If you want, we can keep it for now though. If there was a compile time option to use a system package we shouldn't do that without a good reason.

Reply:
@SuperSandro2000, is it generally better using a pre-compiled bundle with the whole source-code (including necessary submodules), instead of using fetchFromGitHub with fetchSubmodules?